rllib/examples/learners/ppo_with_torch_lr_schedulers.py

```diff
@@ -0,0 +1,147 @@
+|-----------------------------+------------+----------------------+--------+
+| PPO_CartPole-v1_7fc44_00000 | TERMINATED | 192.168.1.178:225070 |     50 |
++-----------------------------+------------+----------------------+--------+
++------------------+------------------------+------------------------+
+|   total time (s) |   num_env_steps_sample |   num_episodes_lifetim |
+|                  |             d_lifetime |                      e |
++------------------+------------------------+------------------------+
+|          59.6542 |                 200000 |                   9952 |
++------------------+------------------------+------------------------+
++------------------------+
+|   num_env_steps_traine |
+|             d_lifetime |
++------------------------|
+|                 210047 |
++------------------------+
+"""
+import functools
+
+from ray.rllib.algorithms.ppo import PPOConfig
+from ray.rllib.utils.metrics import (
+    ENV_RUNNER_RESULTS,
+    EPISODE_RETURN_MEAN,
+    EVALUATION_RESULTS,
+    NUM_ENV_STEPS_SAMPLED_LIFETIME,
+)
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.test_utils import add_rllib_example_script_args
+
+torch, _ = try_import_torch()
+
+parser = add_rllib_example_script_args(default_reward=450.0, default_timesteps=200000)
+parser.set_defaults(enable_new_api_stack=True)
+parser.add_argument(
+    "--lr-const-factor",
+    type=float,
+    default=0.1,
+    help="The factor by which the learning rate should be multiplied.",
+)
+parser.add_argument(
+    "--lr-const-iters",
+    type=int,
+    default=10,
+    help=(
+        "The number of iterations by which the learning rate should be "
+        "multiplied by the factor."
+    ),
+)
+parser.add_argument(
+    "--lr-exp-decay",
+    type=float,
+    default=0.3,
+    help="The rate by which the learning rate should exponentially decay.",
+)
+
+if __name__ == "__main__":
+    # Use `parser` to add your own custom command line options to this script
+    # and (if needed) use their values toset up `config` below.
+    args = parser.parse_args()
+
+    config = (
+        PPOConfig()
+        .environment("CartPole-v1")
+        .training(
+            lr=0.0003,
+            num_sgd_iter=6,
+            vf_loss_coeff=0.01,
+        )
+        .rl_module(
+            model_config_dict={
+                "fcnet_hiddens": [32],
+                "fcnet_activation": "linear",
+                "vf_share_layers": True,
+            }
+        )
+        .evaluation(
+            evaluation_num_env_runners=1,
+            evaluation_interval=1,
+            evaluation_parallel_to_training=True,
+            evaluation_config=PPOConfig.overrides(exploration=False),
+        )
+        .experimental(
+            # Add tow learning rate schedulers to be applied in sequence.
+            _torch_lr_scheduler_classes=[
+                # Multiplies the learning rate by a factor of 0.1 for 10 iterations.
```

Comment:
As a user of this script (who is too lazy reading through all the torch docs :D ) I have a few questions here that we should briefly answer here:
* What is the actually resulting total schedule here if user configured `config.training(lr=L)`? I'm assuming: For `lr_const_iters` iterations: Use `L * 0.1`, after that, jump back up to L, then decay L by lr_exp_decay each iter? So `L *= 0.3` per iter?
* What is an `iter` here? It's not the same necessarily as RLlib algorithm iters, correct, but actually refers to `Learner.update_from...` calls, correct?

Reply:
Great questions! I try to answer them in the following:

- So any list of learning rate schedulers will be chained, i.e. we apply the first and then the second in each iteration. Your assumption is (almost) correct. So the first scheduler will multiply the actual learning rate `L` by `0.1` unless we have stepped 10 times (after this correct, we stay at `L`), i.e `L * 0.1`. The second scheduler then takes this rate (`L * 0.1`) and decays it with a rate of `0.3`, i.e `L = (L * 0.1)  * 0.3`. Because of this latter assignment, the actual learning rate in the second iteration is `(L * 0.1) * 0.3`
- So in the second iteration we will have `(L * 0.1) * 0.3^2` and in the third `(L * 0.1) * 0.3^3` and so on.
- In the 10th iteration however the `ConstantLR` multiplication is off and the learning rate AT THIS POINT is multiplied by the reverse factor, i.e. `(L * 0.1) * 0.3^10 * 1/0.1`.

Yes, it is complex, but this is what we want to offer users. How `torch.rl_scheduler` instances work together is a torch thing and users have to figure it out, we just apply it.